dmcontent/questions.py

```diff
@@ -478,7 +481,19 @@ def get_data(self, form_data):
         this question (therefore only those pairs relevant to this question).
         Filter 0/ False values here and replace with None as they are handled with the optional flag.
         """
-        return {key: form_data[key] or None for key in self.fields.values() if key in form_data}
+        question_data = {key: form_data[key] or None for key in self.fields.values() if key in form_data}
+
+        if self.decimal_place_restriction:
+            # Permissive validation to transform value to 2dp if value contains a `.` but no decimal values or is 1dp
+            pattern_1dp = re.compile(r".*(\.)(?:\d)$")
+            for key, value in question_data.items():
+                if value[-1] == '.':
+                    question_data[key] = value + "00"
+
+                if pattern_1dp.search(value):
```

Comment:
We don't need a compiled pattern here; we could just inline it as `if re.match(r".*(\.)(?:\d)$", value):`

Doesn't _really_ matter much either way, of course...